x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/BlockFactory.java

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.compute.data;
+
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.common.breaker.CircuitBreaker;
+import org.elasticsearch.common.breaker.CircuitBreakingException;
+import org.elasticsearch.common.breaker.NoopCircuitBreaker;
+import org.elasticsearch.common.util.BigArrays;
+import org.elasticsearch.common.util.BytesRefArray;
+import org.elasticsearch.compute.data.Block.MvOrdering;
+
+import java.util.BitSet;
+import java.util.List;
+import java.util.ServiceLoader;
+
+public class BlockFactory {
+
+    private static final BlockFactory NON_BREAKING = BlockFactory.getInstance(
+        new NoopCircuitBreaker("noop-esql-breaker"),
+        BigArrays.NON_RECYCLING_INSTANCE
+    );
+
+    private static final BlockFactory GLOBAL = loadGlobalFactory();
+    // new BlockFactory(new NoopCircuitBreaker("esql_noop_breaker"), BigArrays.NON_RECYCLING_INSTANCE);
```

Comment:
I think I'd prefer this not be static and live on `DriverContext`. We're already passing that in most places, right? In that case I don't think we need the `ServiceLoader` or anything like that - just pass the breaker in. Right?

Reply:
Right.